### PR TITLE
EW-641: Use German terminology in person module.

### PR DESCRIPTION
**Pull request by mkreuzkam-cap**
# Description
Replaces English terminology with German to conform to DDD.
(includes the fields in the person entity)

## Links to Tickets or other pull requests
https://ticketsystem.dbildungscloud.de/browse/EW-641

## Approval for review
- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

### CODE REVIEW

**Review by psachmann — CHANGES_REQUESTED** [review]
- move create-person.dto.ts into the api folder
- cleanup the person-persistence.mapper.profile.ts

**Review by psachmann — COMMENTED** [review]
- move create-person.dto.ts into the api folder
- cleanup the person-persistence.mapper.profile.ts

File: src/modules/person/api/create-person.body.params.ts
Comment (SimoneRadtke-Cap) [inline src/modules/person/api/create-person.body.params.ts:1]: Please change PersonBirthParams required to false and add the type to the ApiProperty

File: src/modules/person/domain/person.do.ts [hunk 1]
```diff
@@ -25,56 +25,56 @@ export class PersonDo<WasPersisted extends boolean> implements DoBase<WasPersist
...
-    public nameTitle?: string;
+    public nameTitel?: string;
 
     @AutoMap(() => [String])
-    public nameSalutation?: string[];
+    public nameAnrede?: string[];
 
     @AutoMap(() => [String])
-    public namePrefix?: string[];
+    public namensPraefix?: string[];
```
Comment (SimoneRadtke-Cap) [inline src/modules/person/domain/person.do.ts:55]: I would prefer 'namePraefix' without 's' here for consistency.

File: src/modules/person/domain/person.do.ts [hunk 2]
```diff
@@ -25,56 +25,56 @@ export class PersonDo<WasPersisted extends boolean> implements DoBase<WasPersist
...
 
     @AutoMap(() => [String])
-    public nameSuffix?: string[];
+    public namensSuffix?: string[];
```
Comment (SimoneRadtke-Cap) [inline src/modules/person/domain/person.do.ts:58]: I would prefer 'nameSuffix' without 's' here for consistency.

File: src/modules/person/api/create-person.body.params.ts
Comment (mkreuzkam-cap) [reply src/modules/person/api/create-person.body.params.ts:1]: Done

File: src/modules/person/api/person-name.params.ts [hunk 3]
```diff
@@ -35,21 +35,21 @@ export class PersonNameParams {
     @IsOptional()
     @IsString()
     @ApiProperty({ name: 'titel', required: false })
```
[the remark this answers is not in the document]
Comment (mkreuzkam-cap) [reply src/modules/person/api/person-name.params.ts]: Done

File: src/modules/person/persistence/person.entity.ts [hunk 4]
```diff
@@ -23,75 +23,75 @@ export class PersonEntity extends TimestampedEntity<PersonEntity, 'id'> {
...
+    @AutoMap(() => [String])
     @Property({ nullable: true, type: ArrayType })
-    public namePrefix?: string[];
+    public namensPraefix?: string[];
 
-    @AutoMap()
+    @AutoMap(() => [String])
     @Property({ nullable: true, type: ArrayType })
-    public nameSuffix?: string[];
+    public namensSuffix?: string[];
```
[the remark this answers is not in the document]
Comment (mkreuzkam-cap) [reply src/modules/person/persistence/person.entity.ts]: Done

File: src/modules/person/api/create-person.body.params.ts [hunk 5]
```diff
@@ -52,9 +52,9 @@ export class CreatePersonBodyParams {
     @AutoMap()
     @IsOptional()
     @IsString()
-    @IsEnum(PersonGender)
-    @ApiProperty({ name: 'geschlecht', enum: PersonGender, required: false })
-    public readonly geschlecht?: PersonGender;
+    @IsEnum(Geschlecht)
+    @ApiProperty({ name: 'geschlecht', enum: Geschlecht, required: false })
```
[the remark this answers is not in the document]
Comment (mkreuzkam-cap) [reply src/modules/person/api/create-person.body.params.ts]: Done

File: src/modules/person/api/person-name.params.ts [hunk 6]
```diff
@@ -5,59 +5,59 @@ import { IsArray, IsOptional, IsString } from 'class-validator';
...
-    @AutoMap()
+    @AutoMap(() => [String])
     @IsOptional()
     @IsArray()
-    @ApiProperty({ name: 'namenssuffix', required: false })
+    @ApiProperty({ type: [String], required: false })
     public readonly namenssuffix?: string[];
 
-    @AutoMap()
+    @AutoMap(() => [String])
```
Comment (SimoneRadtke-Cap) [inline src/modules/person/api/person-name.params.ts:52]: Can you add required false to the ApiProperty?

**Review by SimoneRadtke-Cap — APPROVED** [review]

File: src/modules/person/api/create-person.body.params.ts [hunk 7]
```diff
@@ -46,32 +46,32 @@ export class CreatePersonBodyParams {
     @AutoMap(() => PersonBirthParams)
     @ValidateNested()
     @Type(() => PersonBirthParams)
-    @ApiProperty({ name: 'geburt', required: true })
-    public readonly geburt!: PersonBirthParams;
+    @ApiProperty({ type: PersonBirthParams, required: false })
+    public readonly geburt?: PersonBirthParams;
 
     @AutoMap()
```
Comment (psachmann) [inline src/modules/person/api/create-person.body.params.ts:52]: Type is missing in the AutoMap decorator.

File: src/modules/person/api/create-person.body.params.ts [hunk 8]
```diff
@@ -46,32 +46,32 @@ export class CreatePersonBodyParams {
...
+    public readonly geschlecht?: Geschlecht;
 
     @AutoMap()
     @IsOptional()
     @IsString()
-    @ApiProperty({ name: 'lokalisierung', default: 'de-DE', required: false })
+    @ApiProperty({ default: 'de-DE', required: false })
     public readonly lokalisierung?: string = 'de-DE';
 
     @AutoMap()
```
Comment (psachmann) [inline src/modules/person/api/create-person.body.params.ts:65]: Type is missing in the AutoMap dcorator.

File: src/modules/person/api/person-api.mapper.profile.spec.ts [hunk 9]
```diff
@@ -187,6 +122,14 @@ describe('PersonApiMapperProfile', () => {
             );
         });
 
+        it('should map PersonenkontextDo without loeschungZeitpunkt to CreatedPersonenkontextDto', () => {
+            const personenkontextDo: PersonenkontextDo<true> = DoFactory.createPersonenkontext(true);
+            personenkontextDo.loeschungZeitpunkt = undefined;
+            expect(() => sut.map(personenkontextDo, PersonenkontextDo, CreatedPersonenkontextDto)).not.toThrowError(
```
Comment (psachmann) [inline src/modules/person/api/person-api.mapper.profile.spec.ts:128]: Missing blank lines for better readabilty.

File: src/modules/person/api/person-birth.params.ts
Comment (psachmann) [inline src/modules/person/api/person-birth.params.ts:1]: Line 7: Type is missing in the AutoMap decorator.

File: src/modules/person/api/person.response.ts [hunk 10]
```diff
@@ -1,64 +1,43 @@
...
     public geschlecht!: string;
 
     @AutoMap()
-    @IsOptional()
-    @IsString()
-    @ApiProperty({ name: 'lokalisierung', required: true })
+    @ApiProperty()
     public lokalisierung!: string;
 
     @AutoMap()
```
Comment (psachmann) [inline src/modules/person/api/person.response.ts:40]: Type is missing in the AutoMap decorator.

**Review by psachmann — CHANGES_REQUESTED** [review]

**Pull request merged**